### ISSUE TITLE
DryIoc.Syntax.Ninject project

### ISSUE THREAD

**Issue by dadhi** [dadhi/DryIoc#326]
Similar to DryIoc.Syntax.Autofac.

**Comment by dadhi** [comment]
## Ninject Module

Reference doc: https://github.com/ninject/Ninject/wiki/Modules-and-the-Kernel#modules

Ninject example:
```cs
public class WarriorModule: NinjectModule
{
    public override void Load() 
    {
        Bind<IWeapon>().To<Sword>();
        Bind<Samurai>().ToSelf().InSingletonScope();
    }
}
```

A similar DryIocModule may be approached like this:
```cs
public abstract class DryIocModule
{
    public IRegistrator R { get; set };
    public abstract void Load(); 
}

public class WarriorModule: DryIocModule
{
    public override void Load() 
    {
        R.Register<IWeapon, Sword>();
        R.Register<Samurai>(Reuse.Singleton);
    }
}

// Registering the modules. You may provide the assemblies instead of implementation types.
container.RegisterMany(
    implTypes: new[] { typeof(WarriorModule, /* others */ ) }
    made: PropertiesAndFields.Of.Name(nameof(DryIocModule.R)));
```

**Comment by sigmarsson** [comment]
I wish to replace my NInject configuration. Can I try DryIocModule for size or do I have to wait for a future release ?

**Comment by dadhi** [comment]
You may do it know. Even better if you have some problems I may help with improvements, or at least track this here for documentation and future improvements.

**Comment by sigmarsson** [comment]
I can't find this package even among pre-releases in Nuget.

**Comment by dadhi** [comment]
Huh, it is not ready yet (I am just starting). 
I mean that you can grab the code from example and start experimenting with it and I may help with questions, etc.

**Comment by sigmarsson** [comment]
No stress. Thanks for offering your help. Please estimate your package release plan and I will test it.

**Comment by sigmarsson** [comment]
Hello @dadhi

4.7 is the one shipping this?

**Comment by dadhi** [comment]
No, there is no specific ETA yet. I am using the Ninject on my production project, so I am still planning to work on this.

**Comment by jainashish1711** [comment]
I need some help/suggestions on how can i make migrate from Ninject which is used in my project to DryIOC for Dotnet Framework. I tried but unable to handle Kernel part.
@dadhi @sigmarsson @wjrogers

**Comment by dadhi** [comment]
@jainashish1711 

What's your setup or the example code?

**Comment by jainashish1711** [comment]
Please find sample Startup.cs class below

public partial class Startup
    {
        public static HttpConfiguration HttpConfiguration;
        public void Configuration(IAppBuilder app)
        {
            app.UseCors(Microsoft.Owin.Cors.CorsOptions.AllowAll);

            var kernel = CreateNinjectKernel();
            app.UseNinjectMiddleware(() => kernel);
            app.UseOwinExceptionHandler();

            ConfigureAuth(app, kernel);
            ConfigureWebApi(app);

            GlobalConfiguration.Configuration.Formatters
                .JsonFormatter.SerializerSettings.ReferenceLoopHandling =
                Newtonsoft.Json.ReferenceLoopHandling.Ignore;
        }

        public IKernel CreateNinjectKernel()
        {
            return new StandardKernel(
                NinjectBaseDomainModule.Instance,
                new NinjectDISDomainModule(),
                new NinjectBaseRepositoryModule(),
                new NinjectDISRepositoryModule(),
                new NinjectWebModule());

        }

        private static void ConfigureWebApi(IAppBuilder app)
        {
            HttpConfiguration = new HttpConfiguration();
            WebApiConfig.Register(HttpConfiguration);
            app.UseNinjectWebApi(HttpConfiguration);
        }
    }

Also having trouble binding for these type of codes:
 Bind<ILockingService>().To<DonationLockingService>().WhenInjectedInto<DonationService>();

**Comment by dadhi** [comment]
Here there is a lot of things to digest because you're using Web extensions.

I won't answer it right away, maybe I look later into it when I need to convert my own project. Or you may dig it yourself, and asking some concrete questions when you stuck.

Regarding `WhenInjectedInto`...

## WhenInjectedInto

The example may be translated like this:

Ninject:
```cs
Bind<IInterface>().To<ClassA>()
    .WhenInjectedInto<ClassB>();
```

DryIoc:
```cs
container.Register<IInterface, ClassA>(
    setup: Setup.With(condition: r => r.Parent.ImplementationType == typeof(ClassB)));
```

**Comment by sigmarsson** [comment]
Hello @dadhi , has it bogged down, cancelled or just starving of human resource ? Will you introduce modules as `NInjectModule` works, so that IoC configurations could be reusable ?

**Comment by sigmarsson** [comment]
@dadhi is it yet on your road map ?

**Comment by dadhi** [comment]
Open question. I am currently not working with NInject myself. So it is hard to get with MVP number of features and the examples to convert.
So any help here is appreciated.
